File: 2015/zig/day1.zig
const std = @import("std");
const io = std.io;
const File = std.fs.File;

pub fn main() !void {
    const file = @embedFile("./day1.txt");
    var floor: i32 = 0;
    var pos: i32 = 0;
    for (file) |c| {
        pos += 1;
        // std.debug.print("{c}\n", .{c});
        if (c == '(') {
            floor += 1;
        } else {
            floor -= 1;
        }
        if (floor == -1) {
            std.debug.print("position: {d}\n", .{pos});
        }
    }

    std.debug.print("floor: {d}\n", .{floor});
}

File: 2015/c/src/day1.c
#include "misc.h"

int main(void)
{
	FILE *file = fopen("data/day1.txt", "r");
        
	int32_t floor = 0, position = 0;
	uint64_t end = get_file_length(file);

	for (uint64_t i = 0; i < end; i++) {
		position++;
		if (fgetc(file) == '(') {
			floor++;
		} else {
			floor--;
		}

		if (floor == -1) {
			printf("%d\n", position);
		}
	}

	printf("floor: %d\n", floor);
	fclose(file);
	return 0;
}
